crates/tools/src/lib.rs

```diff
@@ -41,3 +60,61 @@ pub fn collect_tests(s: &str) -> Vec<(usize, Test)> {
     }
     res
 }
+
+
+pub fn update(path: &Path, contents: &str, verify: bool) -> Result<()> {
```

Comment:
Should I change the argument to take `PathBuf` instead of `&Path` as we are using `PathBuf`s anyways?

Reply:
No, by default one should use `&Path`, and switch to `PathBuf` or `&mut PathBuf` only when necessary.

It's the same with Vec, if you can, take `&[T]` as an arg, use `Vec<T>` if you must.